docs/release-process.md

```diff
@@ -24,22 +24,60 @@ Example workflow (see issue process section for more detail [TBD]):
 
 ### Steps to create a release.
 
-1. Create a release branch from main, use the naming format: release-MAJOR.MINOR.
+NGINX Kubernetes Gateway is a trunk based development project; generally following a modified "Branch for release" pattern. Branches will be created as late as possible, often only when a patch is required and the release will first diverge from the main trunk. This results in tags created for planned releases, but branches for patch releases.
 
-2. Create a release candidate tag, use the naming format: vMAJOR.MINOR.PATCH-rc.N (N must start from 1 and monotonically increase with each release candidate).
+#### Planned releases
 
-3. Test the release candidate.
+1. When all committed items are complete within a milestone a release will be created.
 
-    If the tests fail
+1. Trunk merges are ceased for testing. Parallel development is free to continue on development branches (branching can occur here, but we will prefer tagging on trunk for planned releases).
+
+1. Trunk is tested using the "edge" container builds.
+
+  If tests fail:
+  
+    - Fix for issue is created.
+
+    - Open PR against main (trunk) branch.
+
+    - New commit and merge will create a new "edge" container.
+
+    - Repeat.
+
+1. Once tests have passed create a tag (in the form vMAJOR.MINOR.PATCH) on the main (trunk) commit SHA. The docker image will automatically be pushed to `ghcr.io/nginxinc/nginx-kubernetes-gateway:MAJOR.MINOR.PATCH` with the docker tag reflecting the planned release version.
```

Comment:
Right now, in the main branch, the YAML manifests and the docs reference the `edge` tag of the image. 
In release branch, the YAML manifests and the docs reference `0.1.0` tag of the image.
That is also documented here https://github.com/nginxinc/nginx-kubernetes-gateway#nginx-kubernetes-gateway-releases  - users are advised to make sure the image and the docs and manifests match

If we start releasing from the main branch, for the release tag, we will need to update the docs and manifests to use the release tag of the image and then right after the release we will need to update the docs and manifests to use the edge tag of the image.

Reply:
Would it make sense to always have the tags and add a note for `edge`?

Reply:
I think it'll be inconvenient for the user.